cpp/src/unary/math_ops.cu

```diff
@@ -209,9 +209,31 @@ struct DeviceRInt {
   }
 };
 
-// bitwise op
+struct DeviceBitCount {
+  template <typename T>
+  int32_t __device__ operator()(T data)
+  {
+    auto constexpr nbits = CHAR_BIT * sizeof(T);
+    if constexpr (nbits <= 32) {
+      return __popc(static_cast<unsigned int>(data));
+    } else if constexpr (nbits <= 64) {
+      return __popcll(static_cast<unsigned long long>(data));
+    } else {
+      // Convert to unsigned of the same width to handle negative numbers correctly as right shift
+      // on unsigned types is always zero fill.
+      using UnsignedT = std::make_unsigned_t<T>;
+      auto x          = static_cast<UnsignedT>(data);
+      int32_t count   = 0;
+      while (x) {
+        count += x & 1;
+        x >>= 1;
```

Comment:
This branch will handle int128 integer, we can invoke two times of `__popcll` to avoid bit to bit counting.
maybe:
```cpp
else if (nbits == 128) {
  uint64_t const* p = reinterpret_cast<uint64_t *>(&data);
  return __popcll(p[0]) + __popcll(p[1]);
} else {
  // ......
}

```

Reply:
https://godbolt.org/z/5YdhWE1qq

even simpler, just use `cuda::std::popcount`.